x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java

```diff
@@ -280,44 +290,79 @@ static IndexResolution mergedMapping(String indexPattern, Map<String, Map<String
                         errorMessage.append(Arrays.toString(type.getValue().indices()));
                     }
 
-                    errorMessage.insert(0,
-                            "[" + indexPattern + "] points to indices with incompatible mappings; " +
-                            "field [" + name + "] is mapped in [" + types.size() + "] different ways: ");
-                }
-                if (errorMessage.length() > 0) {
-                    return IndexResolution.invalid(errorMessage.toString());
-                }
-                
-                FieldCapabilities fieldCap = types.values().iterator().next();
-                // validate search/agg-able
-                if (fieldCap.isAggregatable() && fieldCap.nonAggregatableIndices() != null) {
-                    errorMessage.append("[" + indexPattern + "] points to indices with incompatible mappings: ");
-                    errorMessage.append("field [" + name + "] is aggregateable except in ");
-                    errorMessage.append(Arrays.toString(fieldCap.nonAggregatableIndices()));
+                    errorMessage.insert(0, "mapped as [" + types.size() + "] incompatible types: ");
```

Comment:
I understand the need to use `types.size()` as it can be of large values when multiple indices are resolved and used in a query, but would you consider at least for smaller size values - like 1,2,3 - to use the word instead? (`one`, `two`, `three`)

Reply:
How would that work? First 1 should never appear so it's about 2, 3 (or whatever limit) and the rest.
This means the message can be "mapped as [three] incompatible types" vs "mapped as [5] incompatible types" since we don't have any utility to convert the numbers into words.
Further more words are never used instead of numbers in any error message that I can think of plus it decreases the readability of the message imo - `[5]` is shorter and more concise than `[five]` which arguably should not have the `[`, `]` around.